core/src/main/kotlin/api/OrganizationsRoute.kt

```diff
@@ -82,13 +84,17 @@ fun Route.organizations() = route("organizations") {
     get(getOrganizations) { _ ->
         val pagingOptions = call.pagingOptions(SortProperty("name", SortDirection.ASCENDING))
 
-        val organizations = organizationService
+        val filteredOrganizations = organizationService
             .listOrganizations(pagingOptions.copy(limit = null, offset = null).mapToModel())
             .filter { hasPermission(it.id, OrganizationPermission.READ) }
-            .paginate(pagingOptions)
+
+        val pagedOrganizations = filteredOrganizations.paginate(pagingOptions)
             .map { it.mapToApi() }
 
-        val pagedResponse = PagedResponse(organizations, pagingOptions)
+        val pagedResponse = PagedResponse2(
+            pagedOrganizations,
+            pagingOptions.toPagingData(filteredOrganizations.size.toLong())
```

Comment:
This converts an `Int`-typed `size` to `Long`. So is there even a point in making the `totalCount` a `Long`?

Reply:
I wonder the same. Especially, as before the whole list of entities is loaded into memory to apply the permissions filter.

Reply:
That's a special case for this endpoint because permission based filtering cannot be done by the database query. All other endpoints (at least currently) will take the total count from the database where it can be a `Long`.

Reply:
> That's a special case for this endpoint because permission based filtering cannot be done by the database query.

Not super deep in how the server does auth checks, but I assume it's possible to get a list of organization ids that the user has the correct roles for in Keycloak? The database query could then be filtered based on this, couldn't it?